python/GafferUI/Timeline.py

```diff
@@ -259,5 +259,34 @@ def __init__( self, value=None, min=0, max=1, hardMin=None, hardMax=None, values
 	def _drawPosition( self, painter, position, highlighted, opacity=1 ) :
 
 		size = self.size()
+
 		# \todo: make sure the TimelineSlider and the AnimationGadget always use the same color
-		painter.fillRect( int(position * size.x), 0, 2, size.y, QtGui.QColor( 240, 220, 40, 255 * opacity ) )
+		color = QtGui.QColor( 240, 220, 40 ) if highlighted else QtGui.QColor( 255, 255, 255 )
+
+		# Note: If you use setPen with a transparent color, the alpha is ignored by drawText
+		painter.setPen( color )
+		painter.setOpacity( opacity )
+
+		sliderX = int(position * size.x)
+
+		# Vertical Line
+
+		painter.fillRect( sliderX, 0, 2, size.y, color )
+
+		# Frame Number
+
+		font = painter.font()
+		font.setPixelSize( 10 )
+		painter.setFont( font )
+
+		frame = self._valueForPosition( position )
+		# Only display decimal places for sub-frame times.
+		frameText = ( "%.0f" if ( abs(frame - round(frame)) < 0.001 ) else "%.2f" ) % frame
+		frameTextSize = QtGui.QFontMetrics( painter.font() ).size( QtCore.Qt.TextSingleLine, frameText )
+
+		textX = sliderX + 6
+		if textX + frameTextSize.width() + 6 >= size.x :
+			textX = sliderX - frameTextSize.width() - 6
```

Comment:
can you explain the +/-6 logic, if any :)
perhaps you could use constant variable to avoid the magic number syndrome?

Reply:
Good call, does https://github.com/GafferHQ/gaffer/pull/3208/commits/e13e52708f582e82200e43b78ba0962b29e37aae help?